### PR TITLE
[clang-tidy][doc][NFC] extract common global options in doc

### PR DESCRIPTION
**Pull request by HerrCai0907**

### CODE REVIEW

File: clang-tools-extra/docs/clang-tidy/GlobalOptions.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+Some options apply to multiple checks. This page lists all the available
+globally options.
+
+.. option:: IncludeStyle
+
+   A string specifying which include-style is used, `llvm` or `google`. Default
+   is `llvm`.
+
+.. option:: IgnoreMacros
```
Comment (carlosgalvezp) [inline clang-tools-extra/docs/clang-tidy/GlobalOptions.rst:13]: I think this one should  _not_ be global. The same is done for clang warnings. Sometimes it may make sense to ignore macros, sometimes not, depending on the check.

File: clang-tools-extra/docs/clang-tidy/GlobalOptions.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+   If set to `true`, the check will not give warnings inside macros. Default
+   is `true`.
+
+.. option:: StrictMode
```
Comment (carlosgalvezp) [inline clang-tools-extra/docs/clang-tidy/GlobalOptions.rst:18]: This option does not have any meaning in this context. Therefore I believe it should be kept local.

File: clang-tools-extra/docs/clang-tidy/checks/cppcoreguidelines/avoid-do-while.rst [hunk 3]
```diff
@@ -26,7 +26,8 @@ Options
 
 .. option:: IgnoreMacros
```
Comment (carlosgalvezp) [inline clang-tools-extra/docs/clang-tidy/checks/cppcoreguidelines/avoid-do-while.rst:28]: Personally I don't see much benefit in this change, it makes the documentation harder to read as one has to go back and forth. I don't see the trouble with this duplication, since it's not really a problem if the sentences don't match exactly with each other - it does not add maintenance cost.

Sometimes DRY can cause more trouble that it's worth - a bit of duplication makes sense in benefit of usability and readability.

**Comment by EugeneZelenko** [comment]
But #113577 is still open. Also local `IncludeStyle` should be just deprecated after.

File: clang-tools-extra/docs/clang-tidy/GlobalOptions.rst (hunk 1 above)
Comment (HerrCai0907) [reply clang-tools-extra/docs/clang-tidy/GlobalOptions.rst:13]: according to current implement of clangtidy, lots of checks will read ignores macros from global scope. i don't think it is a good idea to change lots of checks behavior.

maybe i can notices something in this file to explain global option doesn't apply from every check?

**Comment by HerrCai0907** [comment]
but in code level, we already use the getLocalOrGlobal. but unfortunately, i don't find any description about it. we should have the relative clean map between what we did in code and what we write in doc.
Those three options is not from my personal preferences. they are from current status of code.

**Comment by HerrCai0907** [comment]
but in code level, we already use the getLocalOrGlobal. but unfortunately, i don't find any description about it. we should have the relative clean map between what we did in code and what we write in doc.
Those three options is not from my personal preferences. they are from current status of code.

**Comment by carlosgalvezp** [comment]
They use getLocalOrGlobal probably by mistake, copy-pasting existing code. We should document what we believe makes sense, and fix the implementation accordingly. The current documentation says these options are check-specific, so users should be using them per-check instead of globally. Thus, we wouldn't break users if we change the implementation. So I would vote for replacing the getLocalOrGlobal with getLocal instead.

If we believe the current implementation is faulty, and document that implementation, then we'll have a much harder time to refactor the implementation to what we want (since by documenting, we "promise" a behavior). 

I remember now that the true global options are documented here:

https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/docs/clang-tidy/index.rst?plain=1#L288

So if we create more global options (like `IncludeStyle`) we should probably put them there.

**Comment by carlosgalvezp** [comment]
It could also be that `getLocalOrGlobal` was added to keep backwards compatibility and never break users, e.g for an option that used to be local but has become global. But that doesn't mean that that's what we actually want users to do.

**Comment by HerrCai0907** [comment]
then I think the only option worth to promote is include style.

File: clang-tools-extra/docs/clang-tidy/index.rst [hunk 4]
```diff
@@ -10,6 +10,7 @@ See also:
    :maxdepth: 1
 
    The list of clang-tidy checks <checks/list>
+   The list of clang-tidy global options <GlobalOptions>
    Clang-tidy IDE/Editor Integrations <Integrations>
    Getting Involved <Contributing>
    External Clang-Tidy Examples <ExternalClang-TidyExamples>
```
Comment (nicovank) [inline clang-tools-extra/docs/clang-tidy/index.rst:16]: Just a drive-by comment, noticed this list sometimes uses clang-tidy, sometimes Clang-Tidy.

**Comment by HerrCai0907** [comment]
This PR is not in the correct way. see https://discourse.llvm.org/t/rfc-global-option-rules-for-clang-tidy/83647

File: clang-tools-extra/docs/clang-tidy/index.rst (hunk 4 above)
Comment (HerrCai0907) [reply clang-tools-extra/docs/clang-tidy/index.rst:16]: I think the root cause is the style of titles. Some titles uppercase each word some not.
It looks like in clang/docs, most of rst files' titles use uppercase each word.
